web/src/main/java/org/mskcc/oncotree/api/TumorTypesTxtApi.java

```diff
@@ -32,8 +36,13 @@
         produces = {"text/plain"},
         method = RequestMethod.GET)
     public ResponseEntity<InputStreamResource> tumorTypesTxtGet(
+        @ApiParam(value = "The version of tumor types. For example, 1, 1.1 Please see GitHub for released versions. ")
+        @RequestParam(value = "version", required = false) String version
     ) {
-        InputStream inputStream = TumorTypesUtil.getTumorTypeInputStream();
+        Map<String, TumorType> tumorTypes = new HashMap<>();
+        Version v = (version == null) ? VersionUtil.getDefaultVersion() : VersionUtil.getVersion(version);
+        tumorTypes = CacheUtil.getOrResetTumorTypesByVersion(v);
```

Comment:
This will soon be eliminated (dropping use of CacheUtil .. or modifying to only cache stable taxonomies from TopBraid.)